client/src/graphql_utils/graphql_utils.js

```diff
@@ -56,32 +56,32 @@ export const query_length_tolerant_fetch = async (uri, options) => {
   // important, this regex lazy matches up to and including FIRST ? occurence, which (in a URI)
   // should be where the query string starts. I've complicated it slightly just in case there's ever a ? IN
   // the query string (well, that'd be an encoding error anyway)
-  const url_encoded_query = uri.replace(/^(.*?)\?/, "");
-
-  const query_string_hash = string_hash(url_encoded_query);
+  const query = options.body;
+  const query_hash = string_hash(query);
 
-  const short_uri = `${await get_api_url()}?v=${sha}&queryHash=${query_string_hash}`;
+  const uriWithVersionAndQueryHash = `${await get_api_url()}?v=${sha}&queryHash=${query_hash}`;
 
   const new_options = {
     ...options,
+    method: "GET",
+    body: undefined,
     headers: {
       ...options.headers,
-      "encoded-compressed-query": compressToBase64(
-        decodeURIComponent(url_encoded_query)
-      ),
+      // "encoded-compressed-query": compressToBase64(query),
+      "gql-query": query,
```

Comment:
Note that I stopped encoding it to header that holds the query. It's less annoying to debug, and as far as I can tell HTTP headers don't need to be encoded. 

That being said, HTTP headers should be ASCII, and might not support UTF8. This might cause an issue when we start using the API for search queries, and someone searches for something with accents. If this causes a bug, we can just `_.deburr` queries. All the searching should ignore diacritics/accents anyway.

Reply:
I believe this, or a similar post, was what informed my previous decision to compress: https://stackoverflow.com/questions/3326210/can-http-headers-be-too-big-for-browsers#:~:text=My%20Conclusion%3A%20If%20you%20want,256KB%20for%20all%20headers%20together
  
Don't know if Firefox still has the same 10KB per-header limit. That's still possibly up to 10k characters of query though, so it _sounds_ like we're totally safe even with batching growing query sizes? I'll try and dig up the old PR where I added this just to check whether I was adding it out of extreme caution, or if I'd actually had issues in the wild.

I do agree it'd be nice to stop it, it really is a huge pain when debugging.

Reply:
Original API GET requests PR: https://github.com/TBS-EACPD/infobase-archive/pull/753

Didn't leave much explanation (either of the problem or solution), possible I never figured out what was actually wrong. Looks like I ruled out the per-header limit being the issue at least. Sounds like certain longer queries were somehow throwing CORs errors, and I don't mention any particular browser so it must have been a fully server side issue. Don't see it happening anywhere now though 🤷 .